keps/1618-optional-gc-of-workloads/README.md

```diff
@@ -0,0 +1,440 @@
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+Based on how the retention policy is configured for finished Workloads, 
+several user stories are proposed below.
+
+#### Story 0 - Retention policy for finished Workloads is misconfigured
+
+If the finished Workloads policy is not configured correctly 
+(the provided value is not a valid `time.Duration`), kueue will fail to start 
+during configuration parsing.
+
+#### Story 1 - Retention policy for finished Workloads is not configured (default)
+
+**This story ensures backward compatibility.**
+
+If the object retention policy section is not configured, 
+or finished Workloads retention is either not configured or set to `null`, 
+Workload objects will not be deleted by kueue.
+
+#### Story 2 - Retention policy for finished Workloads is configured and set to a value > 0s
+
+If the object retention policy is configured to a value greater than 0 seconds, 
+the finished Workload retention policy will be evaluated against every 
+completed Workload. The flow is as follows:
+
+1. During kueue's initial reconciliation loop, all previously finished Workloads 
+will be evaluated against the Workload retention policy. They will either be 
+deleted immediately if they are already expired or requeued for reconciliation 
+after their retention period has expired.
+
+2. During subsequent reconciliation loops, each Workload will be evaluated 
+using the same approach as in step 1. However, the evaluation will occur 
+during the next reconciliation loop after the Workload was declared finished. 
+Based on the evaluation result against the retention policy, the Workload will 
+either be requeued for later reconciliation or deleted.
+
+#### Story 3 - Retention policy for finished Workloads is configured and set to 0s
+
+When the object retention policy is set to 0 seconds, finished Workloads will be 
+deleted immediately during the first reconciliation loop after they are declared finished.
+
+### Notes/Constraints/Caveats
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Initially, other naming was considered for config keys. Namely, instead of "Objects," 
+the word "Resources" was used, but based on feedback from `@alculquicondor`, it was changed 
+because "Resources" bears too many meanings in the context of kueue 
+(Kubernetes resources, physical resources, etc.).
+- A different implementation was proposed initially. The deletion logic was not 
+supposed to be part of the `Reconcile` function but rather was supposed to have 
+its own handler/watcher pair. During implementation, this proved challenging 
+without substantial changes to the code. Also, creating a watcher for 
+completed Workloads proved to be very complex.
```

Comment:
Could we move this point to the Alternatives section, and list its pros and cons?

Reply:
Done in: https://github.com/kubernetes-sigs/kueue/pull/2742/commits/2d59922d3a0a32aaea36e70c14d8a2d9f55fd0bc